src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2638,6 +2638,50 @@ theorem getElem_rotateLeft {x : BitVec w} {r i : Nat} (h : i < w) :
       if h' : i < r % w then x[(w - (r % w) + i)] else x[i - (r % w)] := by
   simp [← BitVec.getLsbD_eq_getElem, h]
 
+theorem add_mod_eq_add_sub {w : Nat} {a b : Nat} (hab_ge : a + b ≥ w) (hab_lt : a + b < 2 * w) :
+    (a + b) % w = a + b -  w := by
+  rw [Nat.mod_eq_sub_mod, Nat.mod_eq_of_lt (by omega)]
+  omega
```

Comment:
```suggestion
theorem add_mod_eq_add_sub {w : Nat} {a b : Nat} (hab_ge : a + b ≥ w) (hab_lt : a + b < 2 * w) :
    (a + b) % w = a + b -  w := by
  rw [Nat.mod_eq_sub_mod, Nat.mod_eq_of_lt (by omega)]
  omega
```

This can be generalized by setting $a + b = x$, and writing the statement accordingly. Also, we should make the theorm `private`.